src/main/java/com/flansmod/common/PlayerData.java

```diff
@@ -131,6 +170,18 @@ public void tick(EntityPlayer player)
 		snapshots[0] = new PlayerSnapshot(player);
 	}
 	
+	public void queueReload(ItemStack gunStack, float reloadTime,
+			World world, Entity entity, IInventory inventory, boolean creative, boolean combineAmmoOnReload, boolean ammoToUpperInventory) {		
+		this.gunToReload = gunStack;
+		queuedReload = new QueuedReload(gunStack, reloadTime, world, entity, inventory, creative, combineAmmoOnReload, ammoToUpperInventory);
+	}
+	
+	public boolean isHoldingGunToReload(EntityPlayer player) {
+    	ItemStack heldItem = player.getHeldItem();
+
+    	return !(heldItem == null || gunToReload.stackTagCompound != heldItem.stackTagCompound);
```

Comment:
Hm... still not sure this will do the full equality. Idk if there's some unique item ID?
If this works, what's the unique thing in their NBT, other than current ammo state?

I think different slot may genuinely work quite well, if someone's inv is wiped or the item is somehow removed, the reload cancelling isn't the end of the world. Still not 100% sure.. there must be a better way?

Reply:
You could maybe compare the item directly, since their memory addresses should match.